examples/ts/proxy-usage.ts

```diff
@@ -0,0 +1,41 @@
+
+import ccxt from '../../js/ccxt.js';
+
+// // AUTO-TRANSPILE //
+
+async function example_proxyUrl () {
+    const myEx = new ccxt.kucoin ();
+    myEx.proxyUrl = 'http://5.75.153.75:8090/proxy.php?url=';
+    console.log (await myEx.fetch ('https://api.ipify.org/'));
+}
+
+async function example_httpProxy () {
+    const myEx = new ccxt.kucoin ();
+    myEx.httpProxy = 'http://5.75.153.75:8888'; // "httpProxy" or "httpsProxy" (depending on your proxy protocol)
+    console.log (await myEx.fetch ('https://api.ipify.org/'));
+}
+
+async function example_socksProxy () {
+    const myEx = new ccxt.kucoin ();
+    myEx.socksProxy = 'socks5://127.0.0.1:1080'; // socks5 or socks5h
+    console.log (await myEx.fetch ('https://api.ipify.org/'));
+}
+
+async function example_webSockets () {
+    const myEx = new ccxt.pro.kucoin ();
+    await myEx.initializeProxies (); // only needed for WEBSOCKETS in JAVASCRIPT, not in PYTHON or PHP
+    myEx.httpProxy = 'http://5.75.153.75:8002'; // "httpProxy" or "httpsProxy" (depending on your proxy protocol)
```

Comment:
second question, should we support also `wssProxy/wsProxy`? HTTP refers to a different protocol so it might be confusing

Reply:
that question was also that bothered me too, but as I've seen that in most cases websockets just used http(s)Proxy modules similarly to REST, i've left them as is.  The thing is that there is no lib (in the langs we have in front) where there is something like "WsProxy" module or alike. they just use "http proxy" for websocket connections. 
So, i thought we don't need to make users to set two proxies like:
```
myEx.httpProxy = 'http://1.2.3..';
myEx.wsProxy = 'http://1.2.3..';
```
to make exchange work.
but lmk your arguments

Reply:
even though they might be the same I think it could be useful to support that nomenclature and we can still validate if only one of the supported proxies (https/wssProxy, etc) is set as we do now

Reply:
on second thought, I think that it would complicate things at some bit, because we will need to parse bunch of protocols, we would need to parse `ws` proxies in REST `fetch` method..
would it be acceptable that for a given moment we skip adding `ws/wss` properties in proxies in this PR and in a near future I'll returned to that?